crates/runtime/src/datafusion.rs

```diff
@@ -267,7 +269,7 @@ impl DataFusion {
         &self,
         sql: &str,
         restricted_sql_options: Option<SQLOptions>,
-    ) -> Result<SendableRecordBatchStream> {
+    ) -> Result<QueryResult> {
```

Comment:
Make this Optional. If it is missing, cache is disabled.

Reply:
Like the proposed approach with Option<RetrievalResult>. Updated